### PR TITLE
Add summary information to Viz' Constraints Report

### LINKED ISSUE #947: Add out of box Constraint generation for type checking a column
## Description

Authors can check that a column is a integer by comparing the total count to a type count, but this is a little awkward. whylogs could provide some easier to use helpers:
* check_column_is_<type>

- [ ] I have reviewed the [Guidelines for Contributing](https://github.com/whylabs/whylogs/blob/mainline/.github/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/whylabs/whylogs/blob/mainline/.github/CODE_OF_CONDUCT.md).

### PR DESCRIPTION
**Pull request by FelipeAdachi**
## Description

Enables summary information of metrics on the Viz Constraints Report when hovering on the Pass/Failed icons. It currently displays information only for single metrics constraints.

You can check by going to one of the updated examples and hover on the status to see the summary information. It should look something like the image below.

Other Changes:
- Added type factory constraints: `is_nullable_datatype`
- Added ReportResult named tuple to define report results
- Updated Constraints Suite and Metric Constraints example notebooks to add latest changes

image

## Changes

- Imperative commit or change title (commit ID) 
  - Add some bullet points to explain the change
- Next commit or change (commit ID)

## Related

Relates to organization/repo#number

Closes organization/repo#number

- [ ] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

File: python/whylogs/core/constraints/metric_constraints.py [hunk 1]
```diff
@@ -129,45 +154,74 @@ def validate(self, profile_view: Optional[DatasetProfileView] = None) -> bool:
...
         for metric_name in metric_names:
             metric_constraint = self.dataset_constraints[metric_name]
-            result = metric_constraint.validate(profile)
+            (result, _) = metric_constraint.validate(profile)
             if not result:
                 logger.info(f"{metric_name} failed on dataset")
                 return False
         return True
 
-    def report(self, profile_view: Optional[DatasetProfileView] = None) -> List[Tuple[str, int, int]]:
```
Comment (jamie256) [inline python/whylogs/core/constraints/metric_constraints.py:145]: Looks great, but can we keep the old report method and its signature, add a new method for the new signature so we don't break existing usage.

Comment (FelipeAdachi) [reply python/whylogs/core/constraints/metric_constraints.py:145]: done on latest commits!
I added a deprecated decorator utility to mark report() as deprecated

**Review by jamie256 — CHANGES_REQUESTED** [review]
This approach and capability look nice, lets make sure we don't break existing integrations.

File: python/whylogs/core/constraints/metric_constraints.py [hunk 2]
```diff
@@ -61,15 +77,23 @@ def _validate_metrics(self, metrics: List[Metric]) -> bool:
                 return False
         return True
 
-    def validate(self, dataset_profile: DatasetProfileView) -> bool:
```
Comment (jamie256) [inline python/whylogs/core/constraints/metric_constraints.py:64]: similarly I think we should have the same return type on the original` validate()` method, and add a new method that returns the Tuple

Comment (FelipeAdachi) [reply python/whylogs/core/constraints/metric_constraints.py:64]: done!

**Review by jamie256 — APPROVED** [review]
LGTM!

**Pull request merged**